src/main/java/com/formulasearchengine/mathmlsim/similarity/SubTreeComparison.java

```diff
@@ -0,0 +1,174 @@
+package com.formulasearchengine.mathmlsim.similarity;
+
+import com.formulasearchengine.mathmlsim.similarity.node.MathNode;
+import com.formulasearchengine.mathmlsim.similarity.node.MathNodeGenerator;
+import com.formulasearchengine.mathmlsim.similarity.result.Match;
+import com.formulasearchengine.mathmltools.mml.CMMLInfo;
+import com.google.common.collect.HashMultiset;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Simplistic approach to find similar subtrees between two math expression trees.
+ * <br/>
+ * The algorithm will use {@link MathNode}s.
+ *
+ * @author Vincent Stange
+ */
+public class SubTreeComparison {
+
+    private final Match.Type type;
+
+    public SubTreeComparison(String type) {
+        this.type = Match.Type.valueOf(type);
+    }
+
+    /**
+     * Get a list of similarities between a reference and comparison document.
+     * This method will take an extra step to convert the CMMLInfo document to a MathNode.
+     *
+     * @param refDoc        Reference CMMLInfo document
+     * @param compDoc       Comparison CMMLInfo document
+     * @param onlyOperators find similarities only between operations, leafs are not checked
+     * @return list of similarities, list can be empty but never null
```

Comment:
Kind of, but it is not a fully supported feature in Java. Some code check libraries will notice certain annotation libraries.